tensorflow_datasets/image/div2k.py

```diff
@@ -0,0 +1,102 @@
+month = {January},
+year = {2019}
+}
+"""
+
+_DESCRIPTION = """
+DIV2K dataset: DIVerse 2K resolution high quality images as used for the challenges @ NTIRE (CVPR 2017 and CVPR 2018) and @ PIRM (ECCV 2018)
+"""
+
+class Div2k(tfds.core.GeneratorBasedBuilder):
+  """DIV2K dataset: DIVerse 2K resolution high quality images"""
+
+  VERSION = tfds.core.Version('1.0.0')
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "image": tfds.features.Image(),
+            "downscale_factor": tfds.features.ClassLabel(
+                names=["1", "2", "3", "4"]),
+            "operator": tfds.features.ClassLabel(
+                names=["none", "bicubic"])
+        }),
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    #note: root_path is not a .zip file, must hardcode paths
+    root_path = "https://data.vision.ee.ethz.ch/cvl/DIV2K/"
+    train_paths = dl_manager.download_and_extract({
+        "HR": root_path + "DIV2K_train_HR.zip",
+        "bicubic": root_path + "DIV2K_train_LR_bicubic.zip",
+        })
+    valid_paths = dl_manager.download_and_extract({
+        "HR": root_path + "DIV2K_valid_HR.zip",
+        "bicubic": root_path + "DIV2K_valid_LR_bicubic.zip",
+        })
+    #note: test and unknown files not provided
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            gen_kwargs={
+                "data_paths": train_paths,
+                "split": "train",
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            gen_kwargs={
+                "data_paths": valid_paths,
+                "split": "valid",
+            },
+        ),
+    ]
+
+  def _generate_examples(self, data_paths, split):
+    """Yields examples."""
+
+    file_range = range(1, 801) if split == "train" else range(801, 901)
+
+    #note: uncomment below line if testing with div2k_test.py
+    #file_range = range(1,2) if split == "train" else range(2,3)
+
+    for i in file_range:
+      file_num = "".join(["0" for _ in range(4 - len(str(i)))]) + str(i)
+      image_path = os.path.join(data_paths["HR"],
+                                "DIV2K_"+split+"_HR", file_num+".png")
+      yield file_num, {
+          "image": image_path,
+          "downscale_factor": 1,
+          "operator": "none",
+      }
+
+    for i in file_range:
+      for j in range(2, 5):
+        file_num = "".join(["0" for _ in range(4 - len(str(i)))]) + str(i)
+        image_path = os.path.join(data_paths["bicubic"],
+                                  "DIV2K_"+split+"_LR_bicubic",
+                                  "X"+str(j), file_num+"x"+str(j)+".png")
+        yield file_num+"x"+str(j), {
+            "image": image_path,
+            "downscale_factor": str(j),
+            "operator": "bicubic",
+        }
```

Comment:
Hey @WilliamHYZhang, thanks for the contribution, however this is exactly not what we wanted.
The generator should return 2 image Pair.
`(Low Resolution Image, Corresponding High Resolution Image)`
For that it will be helpful if you use "heavy" configuration for loading the dataset.
Reference: https://www.tensorflow.org/datasets/add_dataset#heavy_configuration_with_builderconfig

Here's a sample which uses Heavy Configuration: https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/audio/librispeech.py

So for example:
```python3
tfds.load("div2k/bicubic_x4")
```
should return the dataset as:
`(Low res image(x4 Bicubic), High Res Image)`
similarly,
```python3
tfds.load("div2k/unkwnown_x2")
```
should return the dataset as:
`(Low res image(x2 Unknown), High Res Image)`

and

```python3
tfds.load("div2k/realistic_mild_x2")
```
should return the dataset as:
`(Low res image(x2 Realistic Mild), High Res Image)`
Hope that makes sense.
Feel free to ask in case if you have any more queries.

Reply:
So sorry about that, I will definitely work on rewriting that.

Reply:
@captain-pool Thanks for your kind feedback, please refer to the new PR(#1276) where I did the changes.